### PR TITLE
Fix BH collapse

### LINKED ISSUE #223: Switching spectral initialization to sklean.manifold.SpectralEmbeddings
A student in our lab is currently looking into spectral initialization, and she found out that `openTSNE.init.spectral(tol=0)` in some cases does not agree to `sklearn.manifold.SpectralEmbedding(affinity='precomputed')`. In some cases it does agree perfectly or near-perfectly, but we have an example when the result is very different, and `SpectralEmbedding` gives what seems like a more meaningful result.

I looked at the math, and it seems that they should conceptually be computing the same thing (`SpectralEmbedding` finds eigenvectors of the L_sym, whereas `init.spectral` finds generalized eigenvectors or W and D, but that should be equivalent, as per https://jlmelville.github.io/smallvis/spectral.html @jlmelville). We don't know what the difference is due to. It may be numerical. 

However, conceptually, it seems sensible if openTSNE would simply outsource the computation to sklearn.

A related issue is that `init.spectral` is not reproducible and gives different results with each run. Apparently the way we initialize `v0` makes ARPACK to still have some randomness. Sklearn gets around this by initializing `v0` differently. I guess openTSNE should do the same -- but of course if we end up simply calling `SpectralEmbedding` then it won't matter.

### PR DESCRIPTION
**Pull request by pavlin-policar**
##### Issue
Fixes #223.

The `is_duplicate` function in the BH implementation had a threshold set to 1e-6, likely taken from other implementations, perhaps scikit-learn. This meant that during the EE phase, where everything was squished together, this condition was triggered a lot, meaning BH wasn't splitting up the points into different quadrants, but was pretending it was dealing with a single point. This mean that there were no repulsive forces being applied to the points at all, leading to an increasing collapse of all the points.

##### Description of changes
I simply changed the threshold for duplicate detection to machine precision, and the issue goes away.
I also copied over the test by @dkobak from #234, but I've also let it run the standard phase. We want to make sure that the embedding isn't collapsed at the end of the optimization, and we don't really care if it is super compressed at the end of the EE phase.

**EDIT**: Not machine precision, because then, duplicate detection actually doesn't work, and the tests fail on iris. Setting the threshold to 1e-16 resolves both issues.

##### Includes
- [X] Code changes
- [X] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by dkobak** [comment]
With all the recent changes it would be great to release a new version (0.6.3 or 0.7), what do you think? Some of the changes (like new learning rate defaults) affect the outcome with default params.

**Comment by pavlin-policar** [comment]
Yes, there have been a few changes since 0.6.2. We've added jittering which actually does change the resulting embeddings a bit, so this probably warrants an 0.7.0 release. The rest was mainly bugfixes I think.

I'd like to get #226 solved first, and there are some issues on conda, but once these get resolved, I'll make a new release.

**Comment by dkobak** [comment]
Yes, init jittering, but also learning rate that is now set to n/exag in each epoch, and momentum fixed to 0.8 throughout.

Cool.
